Tasks/SqlDacpacDeploymentOnMachineGroup/ExecuteScriptsInTransaction.sql

```diff
@@ -0,0 +1,282 @@
+   DROP VIEW _vw_tmpSqlFilesTable
+END
+DECLARE @_viewCommand NVARCHAR(MAX)
+SET @_viewCommand = 'CREATE VIEW _vw_tmpSqlFilesTable AS SELECT t.SqlScriptContent from _tmpSqlFilesTable t'
+EXEC sp_executesql @_viewCommand
+
+DECLARE @_fileList NVARCHAR(MAX)
+DECLARE @_fileName NVARCHAR(MAX)
+DECLARE @_pos INT
+DECLARE @_insertFileCommand NVARCHAR(MAX)
+
+SET @_fileList = '$(_fileList)'
+WHILE LEN(@_fileList) > 0
+BEGIN
+	SET @_pos = CHARINDEX(';', @_fileList)
+	IF (@_pos = 0)
+	BEGIN
+		SET @_fileName = @_fileList
+		SET @_fileName = LTRIM(@_fileName)
+		SET @_fileName = RTRIM(@_fileName)
+		IF (LEN(@_fileName) > 0)
+		BEGIN
+			SET @_insertFileCommand = 'BULK INSERT _vw_tmpSqlFilesTable from ''' + @_fileName + ''' with (ROWTERMINATOR = ''\0'')'
+			EXEC sp_executesql @_insertFileCommand
+		END
+		SET @_fileList = ''
+	END
+	ELSE
+	BEGIN
+		SET @_fileName = SUBSTRING(@_fileList, 1, @_pos - 1)
+		SET @_fileName = LTRIM(@_fileName)
+		SET @_fileName = RTRIM(@_fileName)
+		IF (LEN(@_fileName) > 0)
+		BEGIN
+			SET @_insertFileCommand = 'BULK INSERT  _vw_tmpSqlFilesTable from ''' + @_fileName + ''' with (ROWTERMINATOR = ''\0'')'
+			EXEC sp_executesql @_insertFileCommand
+		END
+		SET @_fileList = SUBSTRING(@_fileList, @_pos + 1, LEN(@_fileList) - @_pos)
+	END
+END
+PRINT 'Completed adding files in temporary table'
+
+PRINT 'Acquiring applock if specified'
+SET @_acquireLock = $(_acquireLockParam)
+SET @_acquireLockMilliseconds = $(_acquireLockMillisecondsParam)
+SET @_acquireLockMaxAttempts = $(_acquireLockMaxAttemptsParam)
+SET @_longRunningThresholdMilliSeconds = $(_longRunningThresholdMilliSecondsParam)
+SET @_lastAttemptsToKillSessions = $(_lastAttemptsToKillSessionsParam)
+SET @_isHosted = $(_isHostedParam)
+
+DECLARE @_batch NVARCHAR(MAX) = ''
+DECLARE @_batchIndex INT
+DECLARE @_statusMessage NVARCHAR(MAX)
+DECLARE @_acquireLockAttemptStart DATETIME
+DECLARE @_acquireLockStart DATETIME
+DECLARE @_milliSecondsUnderLock INT
+DECLARE @_maxMilliSecondsUnderLock INT
+DECLARE @_batchStartTime DATETIME
+DECLARE @_batchMessage VARCHAR(100)
+
+SET @_statusMessage = '@SPID=' + CONVERT(VARCHAR(6), @@SPID)
+RAISERROR (@_statusMessage, 0, 231) WITH NOWAIT
+
+SELECT  @_acquireLock = COALESCE(@_acquireLock, '')
+
+IF (@_acquireLock <> '')
+BEGIN
+    DECLARE @_result                INT = -1
+    DECLARE @_attempt               INT = 1
+    DECLARE @_noWaitAttempt         INT = 1
+    DECLARE @_shortLockName         NVARCHAR(32) = @_acquireLock
+
+    SET @_statusMessage = 'Trying to Acquire Lock #1:' + CONVERT(VARCHAR, GETUTCDATE(), 109)
+    PRINT @_statusMessage
+
+    SET @_acquireLockAttemptStart = GETUTCDATE()
+    
+    BEGIN TRAN
+
+	EXEC @_result = sp_getapplock  @Resource = @_acquireLock, @LockMode = 'Exclusive', @LockTimeout = 0
+
+	IF @_result < 0
+    BEGIN
+		WHILE @_attempt <= @_acquireLockMaxAttempts
+		BEGIN
+        
+			SET @_statusMessage = ''
+
+			IF @_isHosted = 1
```

Comment:
>IF @_isHosted = 1 [](start = 3, length = 17)

Is it about TFS is hosted or on-premise?
If yes, how does this matter when user targets his SQL Server with our task?